PerimeterXModule/PxModule.cs

```diff
@@ -390,9 +420,10 @@ private void VerifyRequest(HttpApplication application)
                     return;
                 }
 
-                // validate using risk cookie
-                IPxCookie pxCookie = PxCookieUtils.BuildCookie(config, pxContext, cookieDecoder);
-                if (!PxCookieValidator.CookieVerify(pxContext, pxCookie))
+				// validate using risk cookie
+				IPxCookie pxCookie = PxCookieUtils.BuildCookie(config, pxContext.PxCookies, cookieDecoder);
+				pxContext.OriginalToken = PxCookieUtils.BuildCookie(config, pxContext.OriginalTokens, cookieDecoder);
```

Comment:
I don't think that it is a good practice to call `PxCookieUtils.BuildCookie` twice just to "prepare" an object (`pxCookie`) that is then used in a single function call (`PxCookieValidator.Verify`).

As such, you should create and use `pxCookie` inside `PxCookieValidator.Verify` (and then pass only the `pxContext` as a parameter).